File: Basic MOBA Mechanics (VR-TP)/ChampionBalance.java

public class Champion {
    private String name;
    private String championClass;
    private String originalLane;
    private String currentLane;
    private int penalty;

    public Champion(String name, String championClass, String originalLane) {
        this.name = name;
        this.championClass = championClass;
        this.originalLane = originalLane;
        this.currentLane = originalLane;
        this.penalty = 0;
    }

    public void moveToLane(String newLane, String currentRole) {
        this.currentLane = newLane;
        applyPenalty(currentRole);
    }

    public void applyPenalty(String currentRole) {
        if (championClass.equals("Juggernaut") && !currentLane.equals(originalLane) && originalLane.equals("Highland")) {
            penalty = 60;
        } else if (championClass.equals("Slayer")) {
            if (originalLane.equals("The Middle Way") && (currentLane.equals("Oblivion") || currentLane.equals("Firestarter"))) {
                penalty = 80;
            } else if (originalLane.equals("Oblivion") && !currentLane.equals("Oblivion")) {
                penalty = 80;
            }
        } else if (championClass.equals("Sniper") && !originalLane.equals("Firestarter")) {
            penalty = 70;
        } else if (championClass.equals("Warlock") && !"Damage".equals(currentRole)) {
            penalty = 70;
        } else if (championClass.equals("Bruiser")) {
            if (originalLane.equals("Highland") && !currentLane.equals("Highland")) {
                penalty = 40;
            } else if (originalLane.equals("Oblivion") && !currentLane.equals("Oblivion")) {
                penalty = 40;
            }
        } else {
            penalty = 0;
        }
    }

    public boolean canUseItem(String itemClass) {
        return championClass.equals(itemClass);
    }

    public String getStatus() {
        return String.format("Nome: %s, Classe: %s, Rota Original: %s, Rota Atual: %s, Penalidade: %d%%",
            name, championClass, originalLane, currentLane, penalty);
    }
}

File: Basic MOBA Mechanics (VR-TP)/ChampionBalance.py

class Champion:
    def __init__(self, name, champion_class, original_lane):
        self.name = name
        self.champion_class = champion_class
        self.original_lane = original_lane
        self.current_lane = original_lane
        self.penalty = 0

    def move_to_lane(self, new_lane, current_role=None):
        self.current_lane = new_lane
        self.apply_penalty(current_role)

    def apply_penalty(self, current_role):
        if self.champion_class == "Juggernaut":
            if self.original_lane == "Highland" and self.current_lane != "Highland":
                self.penalty = 60
        elif self.champion_class == "Slayer":
            if self.original_lane == "The Middle Way" and self.current_lane in ["Oblivion", "Firestarter"]:
                self.penalty = 80
            elif self.original_lane == "Oblivion" and self.current_lane != "Oblivion":
                self.penalty = 80
        elif self.champion_class == "Sniper":
            if self.original_lane != "Firestarter":
                self.penalty = 70
        elif self.champion_class == "Warlock":
            if current_role != "Damage":
                self.penalty = 70
        elif self.champion_class == "Bruiser":
            if self.original_lane == "Highland" and self.current_lane != "Highland":
                self.penalty = 40
            elif self.original_lane == "Oblivion" and self.current_lane != "Oblivion":
                self.penalty = 40
        else:
            self.penalty = 0

    def can_use_item(self, item_class):
        return self.champion_class == item_class

    def status(self):
        return {
            "Nome": self.name,
            "Classe": self.champion_class,
            "Rota Original": self.original_lane,
            "Rota Atual": self.current_lane,
            "Penalidade (%)": self.penalty
        }
